inference/hpu_deepspeed_predictor.py

```diff
@@ -0,0 +1,208 @@
+# Reference: https://github.com/huggingface/optimum-habana/blob/main/examples/text-generation/utils.py
```

Comment:
can you follow our openAI API code and use the same way to write the license and header here?